sdb/commands/zfs/histograms.py

```diff
@@ -0,0 +1,149 @@
+              16.0KB:   4876 *********************
+              32.0KB:   9138 ****************************************
+              64.0KB:   4508 ********************
+             128.0KB:   2783 ************
+             256.0KB:   1952 *********
+             512.0KB:   1218 *****
+               1.0MB:    675 ***
+               2.0MB:    486 **
+               4.0MB:    267 *
+               8.0MB:    110
+              16.0MB:     50
+              32.0MB:     18
+              64.0MB:      8
+             128.0MB:     11
+             256.0MB:    102
+            Histogram Median: 339.7MB
+    """
+
+    names = ["zfs_histogram", "zhist"]
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument("offset", nargs="?", default=0, type=int)
+        return parser
+
+    @staticmethod
+    def histogram_median(hist: drgn.Object, offset: int = 0) -> int:
+        """
+        Returns the approximated median of a ZFS histogram.
+        """
+        canonical_type = sdb.type_canonicalize(hist.type_)
+        assert canonical_type.kind == drgn.TypeKind.ARRAY
+        assert sdb.type_canonicalize(
+            canonical_type.type).kind == drgn.TypeKind.INT
+
+        total_space = 0
+        for (bucket, value) in enumerate(hist):
+            total_space += int(value) << (bucket + offset)
+
+        if total_space == 0:
+            return 0
+
+        space_left, median = total_space / 2, 0
+        for (bucket, value) in enumerate(hist):
+            space_in_bucket = int(value) << (bucket + offset)
+            if space_left <= space_in_bucket:
+                median = 1 << (bucket + offset)
+                #
+                # Size of segments may vary within one bucket thus we
+                # attempt to approximate the median by looking at the
+                # number of segments in the bucket and assuming that
+                # they are evenly distributed along the bucket's range.
+                #
+                bucket_fill = space_left / space_in_bucket
+                median += round(median * bucket_fill)
+                break
+            space_left -= space_in_bucket
+        return median
+
+    @staticmethod
+    def print_histogram_median(hist: drgn.Object,
+                               offset: int = 0,
+                               indent: int = 0) -> None:
+        median = ZFSHistogram.histogram_median(hist, offset)
+        print(f'{" " * indent}Histogram Median: {fmt.size_nicenum(median)}')
+
+    @staticmethod
+    def print_histogram(hist: drgn.Object,
+                        offset: int = 0,
+                        indent: int = 0) -> None:
+        canonical_type = sdb.type_canonicalize(hist.type_)
+        assert canonical_type.kind == drgn.TypeKind.ARRAY
+        assert sdb.type_canonicalize(
+            canonical_type.type).kind == drgn.TypeKind.INT
+
+        max_ = 0
```

Comment:
[nit] I assume the trailing underscore is to avoid a conflict with the built-in `max()` function.  It might be cleaner to do that by renaming this to `max_value` or `max_count`